Tests/SwiftDocCTests/Infrastructure/DocumentationCuratorTests.swift

```diff
@@ -279,6 +279,10 @@ class DocumentationCuratorTests: XCTestCase {
             ### Extraneous list item content
             - <doc:MyKit/MyClass>.
             - <doc:MyKit/MyClass> ![](featured.png) and *more* content...
+            - <doc:MyKit/MyClass> @Comment { This is an invalid comment directive }
```

Comment:
Maybe we can add an explanation here? Ideally this would work – so good to explain why it doesn't.

```suggestion
            - <doc:MyKit/MyClass> @Comment { We (unfortunately) expect a warning here because DocC doesn't support directives in the middle of a line. }
```

Reply:
Couldn't we make this work if the `contains(where:)` closure checked for block directives?
```swift
if let directive = child as? BlockDirective {
    return directive.name == "Comment"
}
```

Reply:
It looks like anything after the first child is parsed as Text, so when I tried that it never found a block directive.

Reply:
Ah. That's too bad.

Reply:
Swift-Markdown only supports parsing directives that start at the beginning of the line unfortunately.